### DISCUSSION TITLE
Extract JPEG data without file output

### DISCUSSION THREAD

**Discussion by mklawler** [photostructure/exiftool-vendored.js#136] [Q&A] [answered]
Would love to be able to get JPEG data from NEF files and use that in-memory data to display in an existing electron window.  Seems like I have to write to file and read from file.

**Comment by mceachen** [comment, marked as answer]
Electron is a **bear** to get packaged cross-platform.

Once it's there, you'll need to do a `.read()` call to see what embedded images are available, and then extract the preview: https://photostructure.github.io/exiftool-vendored.js/#md:extracting-embedded-images
